codegen/src/generator/json.rs

```diff
@@ -0,0 +1,77 @@
+use inflector::Inflector;
+
+use botocore::{Operation, Service};
+use super::GenerateProtocol;
+
+pub struct JsonGenerator;
+
+impl GenerateProtocol for JsonGenerator {
+    fn generate_methods(&self, service: &Service) -> String {
+        service.operations.values().map(|operation| {
+            let output_type = operation.output_shape_or("()");
+
+            format!("
+                {documentation}
+                pub fn {method_name}(&mut self, input: &{input_type}) -> AwsResult<{output_type}> {{
+                    let encoded = serde_json::to_string(input).unwrap();
+                    let mut request = SignedRequest::new(\"{http_method}\", \"{endpoint_prefix}\", &self.region, \"{request_uri}\");
+                    request.set_content_type(\"application/x-amz-json-1.0\".to_owned());
```

Comment:
I think this is a regression, we need 1.1, not 1.0.  Will fix up shortly after testing.